lib/plugins/inventory.js

```diff
@@ -427,6 +432,12 @@ function inject (bot, { hideErrors }) {
         }
       }
       await Promise.all(toUpdate)
+
+      if (slot === 2 && !window.selectedTrade.tradeDisabled && expectTradeUpdate(window)) {
+        // After the trade goes through, if the inputs are still satisfied,
+        // expect another update in slot 2
+        await once(bot.currentWindow, 'updateSlot:2')
+      }
```

Comment:
In the trade loop, if the requirements of a trade are still satisfied after `putAway` but the window doesn't have an item in slot 2, then `putRequirements` will not deposit anything, an update in slot 2 will not be awaited, and `putAway` will be immediately called again without anything in slot 2. So here, we should only consider the trading window "fully updated" if the state of slot 2 correctly reflects whether the trade inputs are satisfied.